crates/app/signing/src/collector/extractor_of_instances_required_to_sign_transactions.rs

```diff
@@ -7,20 +7,18 @@ impl ExtractorOfInstancesRequiredToSignTransactions {
     /// Extracts factor instances required to sign transactions.
     /// Returns a set of `HierarchicalDeterministicFactorInstance`.
     /// Returns an error if the `SignaturesCollectorPreprocessor` fails to initialize.
-    pub fn extract<S: Signable>(
-        profile: &Profile,
+    pub fn extract<S: Signable, P: GetEntityByAddress + HasFactorSources>(
```

Comment:
This allows us to not depend on Profile crate! which is important tactics in general - so we can have a "Broad dependency tree" instead of having a "long tree", by describe ONLY what is needed (with traits) .